lib/livebook/ecto_types/hex_color.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Livebook.EctoTypes.HexColor do
+  @moduledoc false
+  use Ecto.Type
+
+  alias Livebook.Utils
+
+  def type, do: :string
+
+  def cast(value), do: validate(value)
+  def load(value), do: validate(value)
+  def dump(value), do: validate(value)
+
+  defp validate(value) do
+    if Utils.valid_hex_color?(value) do
+      {:ok, value}
+    else
+      {:error, message: "not a valid color"}
+    end
+  end
```

Comment:
```suggestion
  def load(value), do: {:ok, value}
  def dump(value), do: {:ok, value}

  def cast(value) do
    if Utils.valid_hex_color?(value) do
      {:ok, value}
    else
      {:error, message: "not a valid color"}
    end
  end
```

Both load and dump assume the data is valid. It is just about conversion from and to the DB, which in our case is a no-op. :)